### PR TITLE
add vxlan-generic

### PR DESCRIPTION
**Pull request by zhangml**
This time pr has been modified as follows:
1. Support vxlan generic type interface.
2. Remove aca_oam_port_manager.
3. Added auxgateway information in vpc_table_entry.

### CODE REVIEW

**Comment by zhangml** [comment]
Hi,Eric. @er1cthe0ne At the end of the test, there will be a segfault after clean up, which is caused by "aca_oam_test.cpp", but I cannot locate the specific cause.
```
  YOU HAVE 9 DISABLED TESTS

g_total_execute_system_time = 10707379 microseconds or 10707 milliseconds
g_total_execute_ovsdb_time = 8309807 microseconds or 8309 milliseconds
g_total_execute_openflow_time = 1267746 microseconds or 1267 milliseconds
g_total_vpcs_table_mutex_time = 2655367 microseconds or 2655 milliseconds
g_total_update_GS_time = 1797638 microseconds or 1797 milliseconds
Program exiting, cleaning up...
Segmentation fault
```

File: src/zeta/aca_oam_server.cpp [hunk 1]
```diff
@@ -262,20 +252,17 @@ int ACA_Oam_Server::_add_direct_path(oam_match match, oam_action action)
...
                      ",nw_dst=" + match.dip + ",tp_src=" + match.sport +
                      ",tp_dst=" + match.dport + ",dl_vlan=" + vlan_id;
-  string cmd_action = ",actions=\"strip_vlan,load:" + match.vni +
-                      "->NXM_NX_TUN_ID[],mod_dl_dst=" + action.inst_dl_dst +
-                      ",mod_nw_dst=" + action.inst_nw_dst +
-                      ",output:" + outport_name + "\"";
+  string cmd_action = ",actions=\"strip_vlan,load:" + to_string(match.vni) +
+                      "->NXM_NX_TUN_ID[],set_field:" + action.node_nw_dst +
+                      "->tun_dst,mod_dl_dst=" + action.inst_dl_dst +
+                      ",mod_nw_dst=" + action.inst_nw_dst + ",output:vxlan-generic\"";
```
Comment (er1cthe0ne) [inline src/zeta/aca_oam_server.cpp:265]: do you need the below? I believe the out going packet already has the correct information. 

mod_dl_dst=" + action.inst_dl_dst + ",mod_nw_dst=" + action.inst_nw_dst

File: src/zeta/aca_zeta_programming.cpp [hunk 2]
```diff
@@ -116,40 +71,38 @@ int ACA_Zeta_Programming::delete_zeta_config(const alcor::schema::AuxGateway cur
...
 {
   unsigned long not_care_culminative_time;
   int overall_rc = EXIT_SUCCESS;
-
+  if (is_exist_group_rule(zeta_cfg->group_id)) {
+    return overall_rc;
+  }
   //adding group table rule
-  string cmd = "-O OpenFlow13 add-group br-tun group_id=" + zeta_cfg->group_id + ",type=select";
+  string cmd = "-O OpenFlow13 add-group br-tun group_id=" + to_string(zeta_cfg->group_id) +
```
Comment (er1cthe0ne) [inline src/zeta/aca_zeta_programming.cpp:100]: please confirm the group rule can send both regular packets and also ARP packets to the Zeta gateway.

File: src/zeta/aca_zeta_programming.cpp [hunk 3]
```diff
@@ -184,6 +137,8 @@ int ACA_Zeta_Programming::_delete_zeta_group_entry(zeta_config *zeta_cfg)
   return overall_rc;
 }
 
+// Determine whether the group table rule already exists?
+// I am not sure if this approach is appropriate?
```
Comment (er1cthe0ne) [inline src/zeta/aca_zeta_programming.cpp:141]: it is a decent approach. Need to think about the case that the group rule is there but it is the exactly as the one we expected, like the bucket destination is wrong. That's an improvement to consider in the future and not a blocker now.

File: include/aca_vlan_manager.h [hunk 4]
```diff
@@ -60,20 +67,22 @@ class ACA_Vlan_Manager {
...
-  int create_neighbor_outport(alcor::schema::NetworkType network_type, string remote_host_ip,
-                              uint tunnel_id, ulong &culminative_time);
-
   int delete_neighbor_outport(string neighbor_id, uint tunnel_id,
                               string outport_name, ulong &culminative_time);
 
   int get_outports_unsafe(uint tunnel_id, string &outports);
 
-  void set_aux_gateway(uint tunnel_id, const string auxGateway_id);
+  void set_auxgateway(uint tunnel_id, const string auxGateway_id, uint oam_port);
```
Comment (er1cthe0ne) [inline include/aca_vlan_manager.h:75]: we are making auxgateway=zeta gateway in the current implementation. auxgateway can include zeta and internet and NAT gateway at the same time. Let's focus the current change for zeta gateway and move all the functions to zeta files.

**Comment by er1cthe0ne** [comment]
> Hi,Eric. @er1cthe0ne At the end of the test, there will be a segfault after clean up, which is caused by "aca_oam_test.cpp", but I cannot locate the specific cause.
> 
> ```
>   YOU HAVE 9 DISABLED TESTS
> 
> g_total_execute_system_time = 10707379 microseconds or 10707 milliseconds
> g_total_execute_ovsdb_time = 8309807 microseconds or 8309 milliseconds
> g_total_execute_openflow_time = 1267746 microseconds or 1267 milliseconds
> g_total_vpcs_table_mutex_time = 2655367 microseconds or 2655 milliseconds
> g_total_update_GS_time = 1797638 microseconds or 1797 milliseconds
> Program exiting, cleaning up...
> Segmentation fault
> ```

perfect time to learn about gdb, it should be able to pinpoint the problem, below is some links to get started:
https://stackoverflow.com/questions/46749834/segmentation-fault-with-gdb-debugger-c
http://unknownroad.com/rtfm/gdbtut/gdbsegfault.html

File: src/zeta/aca_oam_server.cpp (hunk 1 above)
Comment (er1cthe0ne) [reply src/zeta/aca_oam_server.cpp:265]: don't worry about it as discussed.

**Comment by er1cthe0ne** [comment]
FYI, I have updated my branch, the traffic test is not working yet, I am still working on it: https://github.com/er1cthe0ne/alcor-control-agent/tree/perf/hashtable

**Comment by er1cthe0ne** [comment]
I saw you have another PR open, please close this one if it is no longer needed.

**Pull request closed without merging**